### ISSUE TITLE
Button colors not working as expected in variant light

### ISSUE THREAD

**Issue by Orhmo** [mantinedev/mantine#4547]
### What package has an issue

@mantine/core

### Describe the bug

The button colors when using variant light do not function as expected. Instead, a light green filled color is shown as the button color, and the variant color inputted in the code is shown on hover. 
Code example: 
 <Button variant="light" color="red">
      Cancel
  </Button>
When a color is added to the filled button. The default fill color is shown instead of the color inputted in the code.

### What version of @mantine/hooks page do you have in package.json?

^6.0.13

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

**Comment by rtivital** [comment]
Please provide a sandbox that reproduces the issue, template – https://codesandbox.io/s/mantine-template-9ze89?file=/src/App.tsx:0-6

**Comment by Orhmo** [comment]
@rtivital  I tried using the template on codesand box , which didn't show the issue I am experiencing. However, I have used the Button feature on two Javascript projects (ReactJs and NextJs) and have experienced the same issue with the button feature as stated initially. Using the inspect tool, it shows that  (.mantine-1johb54) has a background color of green which overrides my color set. However, no green background color was set.
